cmd/nuntium/mediator.go

```diff
@@ -430,29 +431,33 @@ func (mediator *Mediator) uploadFile(filePath string) (string, error) {
 	return mSendRespFile, uploadErr
 }
 
-// by default this method returns true, unless it is strictly requested to disable
+// By default this method returns true, unless it is strictly requested to disable.
 func mmsEnabled() bool {
 	conn, err := dbus.Connect(dbus.SystemBus)
 	if err != nil {
+		log.Printf("mmsEnabled: connecting to dbus failed: %v", err)
 		return true
 	}
+
 	usr, err := user.Current()
 	if err != nil {
+		log.Printf("mmsEnabled: getting user failed: %v", err)
 		return true
 	}
 	activeUser := dbus.ObjectPath("/org/freedesktop/Accounts/User" + usr.Uid)
 
 	obj := conn.Object("org.freedesktop.Accounts", activeUser)
-
 	reply, err := obj.Call("org.freedesktop.DBus.Properties", "Get", "com.ubuntu.touch.AccountsService.Phone", "MmsEnabled")
 	if err != nil || reply.Type == dbus.TypeError {
-		fmt.Println(fmt.Errorf("failed to get mms option: %s", err))
+		log.Printf("mmsEnabled: failed to get mms option: %v", err)
 		return true
 	}
+
 	mms := dbus.Variant{true}
-	if err := reply.Args(&mms); err == nil {
-		return mms.Value.(bool)
+	if err := reply.Args(&mms); err != nil {
+		log.Printf("mmsEnabled: failed to get mms option reply: %v", err)
+		return true
 	}
 
-	return true
+	return mms.Value.(bool)
```

Comment:
Why is the return value logic reversed between the if statement above and here, in this commit? Can you also please describe in the PR description how to test this change? Thanks.

Reply:
So it is semantically consistent. See this example:
```go
func someFunc() bool {
  err:= someCommand()
  if err != nil { //early return
    return true
  }

  value, err := someOtherCommand()
  if err == nil {
    return value
  }

  return true
}
```
Like this it's confusing. Best practice is make early return on errors in conditions and value return at end. Like this:
```go
func someFunc() bool {
  err:= someCommand()
  if err != nil { //early return
    return true
  }

  value, err := someOtherCommand()
  if err != nil { //early return
    return true
  }

  return value
}
```

Its logicaly the same, but less confusing. And that's what I've done in this PR. There is really nothing to test.
But if someone wants to test it, then the test is the practically the same as for previous PR #6 . If you disable MMS somewhere in settings, you should not receive MMS sent to you (except there should be "MMS is disabled" in logs).

Reply:
Ah, I see. I missed the `!` change there.